jekyll/_cci2/contexts.md

```diff
@@ -49,8 +49,57 @@ workflows:
+
+## Running Workflows with a Restricted Context
+
+To invoke a workflow that uses a restricted context, a user must be a member of one of the security groups for the context or the workflow will fail with the status of `Unauthorized`. If you add a context to your workflow and you are **not** a member of any of the context groups, the workflow will fail as `Unauthorized`.
+
+**Note:** Bitbucket repositories do **not** provide an API that allows CircleCI contexts to be restricted, only GitHub hosted projects or projects that use GitHub Enterprise with CircleCI include the ability to restrict contexts with security groups.
+
+### Restrict a Context to a Security Group or Groups
+
+You must be an Org Admin to complete the following task.
+
+1. Navigate to Organization Settings > Contexts page in the CircleCI app. The list of contexts appears. The default security group is `organization` and allows all users in the org to invoke jobs with that context.
+2. Click the Create Context button if you will use a new context or click the name of an existing context.
+3. Click the Add Group link. The Add Groups dialog box appears.
+4. Select GitHub teams or LDAP groups to add to the context and click the Add button. Use of the context is now limited to the selected groups.
+5. Click Add Environment Variables to add environment variables to the context if none exist and click the Add button. Use of the environment variables for this context is now limited to members of the security groups.
+6. Navigate to Organization Settings > Contexts in the CircleCI app. The security groups appear in the Security column for the context.
+
+Only members of the selected groups may now use the context in their workflows or add or remove environment variables for the context. 
+
+### Approving Jobs that use Restricted Contexts
+
+Adding an approval job to a workflow allows a user to manually approve the use of a restricted context. 
+
+To restrict running of jobs that are downstream from an approval job, add a restricted context to those downstream jobs. 
+
+For example, if you want the execution of job C and job D restricted to a secutiry group, you need to add an approval job B before the jobs C and D to that use a context with a security group. That is, you can have four jobs in a workflow, job A can run unrestricted, the approval job B may be approved by any member, but the jobs C and D after the approval may only be executed by someone in the security group for the context used on jobs C and D. 
+
+If the approver of a job is not part of the restricted context, it is possible to approve the job B, however the jobs C and D in the workflow will fail as unauthorized. That is, the Approval button will appear for every user, even for users who are not part of the group with permissions for the context. When the downstream jobs fail with Unauthorized, it indicates an approval was made by a user who is not part of the security group for the downstream jobs. 
```

Comment:
This UI appears to violate the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment). Say I'm not in the restricted list, how can I know that I should not try to approve but find someone who can? Do I need to try it, fail, and then try to get someone else to re-run it from failed for me? (And how can I find out who is in the group?)

Would you consider allowing the restricted context to work for `type: approval` jobs too, so people can model a better UX than described here? I get that the restriction has to be for using the context, otherwise people could just edit the workflow to remove the approval.